File: src/main/java/core/procedures/vectors/VectorFill.java
package core.procedures.vectors;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;
import core.scm.MutableVector;
import core.scm.Void;

import java.util.Arrays;

public final class VectorFill extends AFn {

  public VectorFill() {
    super(new FnArgsBuilder().min(2).max(2).mandatory(new Class[]{MutableVector.class, Object.class}).build());
  }

  @Override
  public String getName() {
    return "vector-fill!";
  }

  @Override
  public Object apply2(Object arg1, Object arg2) {
    MutableVector vector = (MutableVector) arg1;
    Arrays.fill(vector.getArray(), arg2);
    return Void.VOID;
  }
}

File: src/main/java/core/procedures/vectors/VectorFill.kt
package core.procedures.vectors

import core.procedures.AFn
import core.procedures.Arity.Exactly
import core.scm.MutableVector

class VectorFill : AFn<Any?, Unit>(name = "vector-fill!", arity = Exactly(2),
                       mandatoryArgsTypes = arrayOf(MutableVector::class.java, Any::class.java)) {

    override operator fun invoke(arg1: Any?, arg2: Any?) = (arg1!! as MutableVector).array.fill(arg2)
}
